dns/query.py

```diff
@@ -37,6 +37,10 @@
 import dns.rdataclass
 import dns.rdatatype
 
+import requests
+import requests.packages.urllib3.util.connection as urllib3_cn
```

Comment:
From experience, that import is extremely brittle.

Reply:
@baloo What would you recommend? You mean the `import as` part, or the fact that I'm importing urllib3.util.connection from requests?

Reply:
urllib3 is vendored in requests because ... reasons. Some distro packages dropped that vendoring (because against policy?).
Some older versions of requests did not have that, and there is no guarantee this stays like that, this is not part of the requests api. 

at least, it would wise to wrap it in a:
``` python
try:
    import requests.packages.urllib3.util.connection as urllib3_cn
except ImportError:
    urllib3_cn = None
```

and get an `if urllib3_cn` logic where consumed.

Might be worth to throw a warning for the next developer that comes by when it fails to import and the af logic does not work anymore (it also gets reported in the tests, and it makes easier to catch an API break in requests).

Reply:
Ah yes, I can see all kind of hairy stuff going on with `request.packages`, especially around May 2017.